competition/src/main/java/frc/robot/subsystem/Intake.java

```diff
@@ -63,4 +93,46 @@ public boolean isDown(){
         return isDown;
     }
 
+    public boolean shouldIntake(){
+        return shouldIntake;
+    }
+
+    public void run(){
+        // detect current ball color
+        double allianceColorConfidence = allianceColorFilter.calculate(colourSensor.isColor(allianceColor) ? 1 : 0);
+        double oppositionColorConfidence = oppositionColorFilter.calculate(colourSensor.isColor(oppositionColor) ? 1 : 0);
```

Comment:
I'd like to keep the intake code as simple as possible, so if we can I'd like to move some of this logic to another class, maybe called `CargoDetector`? Maybe have methods `.seesRed()`, `.seesBlue()` and `.seesChange()`? Also, The median filter alone if it goes above, below and above the confidence threshold very quickly. I would add another threshold for when a color is no longer detected, like `.5`.